crates/data-router/src/handler.rs

```diff
@@ -204,17 +191,15 @@ async fn route_static(
     Ok(())
 }
 
-#[tracing::instrument(skip(publisher))]
+#[tracing::instrument(skip(cache, publisher))]
 async fn route(
-    cache: &Mutex<LruCache<Uuid, String>>,
+    cache: Arc<Mutex<DynamicCache<Uuid, String>>>,
```

Comment:
```suggestion
    cache: &Mutex<DynamicCache<Uuid, String>>,
```
plus replacing ` route(self.cache.clone(),` to `route(&self.cache,` in two places.
We don't have to clone `Arc`. Not much but this saves us from unnecessary synchronization between different threads(cloning Arc).

Reply:
Tbh. I think we can safely remove `Arc` now - it was needed earlier, but I don't see any more of it being cloned.